### PR TITLE
[MU4] Fixed typing text characters twice when screen reader is running

### PR DESCRIPTION
**Pull request by Eism**
Resolves: #10467

### CODE REVIEW

File: src/notation/internal/notationinteraction.cpp [hunk 1]
```diff
@@ -2771,6 +2771,41 @@ void NotationInteraction::changeEditElement(EngravingItem* newElement)
...
+
+    if (m_editData.element->isEditAllowed(m_editData)) {
+        return true;
+    }
+
+    if (event->modifiers() & Qt::KeyboardModifier::AltModifier) {
+        return false;
+    }
+
+    if (m_editData.element->isTextBase()) {
```
Comment (RomanPudashkin) [inline src/notation/internal/notationinteraction.cpp:2791]: This code looks strange. Is it a typo?

File: src/notation/internal/notationinteraction.cpp [hunk 2]
```diff
@@ -2771,6 +2771,41 @@ void NotationInteraction::changeEditElement(EngravingItem* newElement)
...
+        return false;
```
Comment (RomanPudashkin) [inline src/notation/internal/notationinteraction.cpp:2792]: This code looks strange. Is it a typo?

Comment (Eism) [reply src/notation/internal/notationinteraction.cpp:2792]: No, why strange?

Comment (RomanPudashkin) [reply src/notation/internal/notationinteraction.cpp:2792]: Looks like we can't edit text elements anymore

Comment (Eism) [reply src/notation/internal/notationinteraction.cpp:2792]: Yes, we should get the answer about the ability to edit the text above:
```
 if (m_editData.element->isEditAllowed(m_editData)) {
        return true;
    }
```

**Comment by Eism** [comment]
@shoogle thanks for testing!
Fixed, it was a typo :)

File: src/notation/internal/notationinteraction.cpp [hunk 3]
```diff
@@ -2771,6 +2771,41 @@ void NotationInteraction::changeEditElement(EngravingItem* newElement)
...
 }
 
+bool NotationInteraction::isEditAllowed(QKeyEvent* event)
+{
+    if (!m_editData.element) {
+        return false;
+    }
+
+    m_editData.modifiers = event->modifiers();
+    m_editData.key = event->key();
```
Comment (RomanPudashkin) [inline src/notation/internal/notationinteraction.cpp:2781]: This method should not change the state of the class

File: src/notation/internal/notationinteraction.cpp [hunk 4]
```diff
@@ -2771,6 +2771,41 @@ void NotationInteraction::changeEditElement(EngravingItem* newElement)
...
+
+    if (event->modifiers() & Qt::KeyboardModifier::AltModifier) {
+        return false;
+    }
+
+    if (m_editData.element->isTextBase()) {
+        return false;
+    }
+
+    QSet<int> navigationKeys = {
```
Comment (RomanPudashkin) [inline src/notation/internal/notationinteraction.cpp:2795]: This set should be static const

**Review by RomanPudashkin — APPROVED** [review]

**Review by shoogle — APPROVED** [review]
@Eism, it works perfectly now, thanks!

**Comment by DmitryArefiev** [comment]
Tested #10467 on LinuxMint 20.2 (Win10, Mac12)  - FIXED      

Please merge into master

**Pull request merged**